app/routers/organizations.py

```diff
@@ -388,3 +391,92 @@ async def delete_organization_by_id(
+    additional_admin_repo: AdditionalAdminRequestRepository,
+    data: AdditionalAdminRequestCreate,
+):
+    """
+    This endpoint adds additional admins to FinOps for Cloud.
+    If the user to be added as an admin does not exist, it will be created and added to
+    the given organization's id, and the reset flow will also be triggered
+    to allow the user to set a new password.
+    """
+    new_user_created = False
+    if db_organization.status == OrganizationStatus.DELETED:
+        logger.info(f"The Organization {db_organization.name} is deleted.")
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"Organization {db_organization.name} is deleted.",
+        )
+    # fetch user by email
+    with wrap_http_error_in_502("Error checking or creating user in FinOps for Cloud"):
+        try:
+            response = await optscale_auth_client.get_existing_user_info(data.email)
+            user_data = response.json()["user_info"]
+        except UserDoesNotExist:
+            logger.info(
+                f"User with email {data.email} does not exist in FinOps for Cloud."
+                f"Going to create it."
+            )
+            # create the user
+            response = await api_modifier_client.create_user(
+                email=data.email,
+                display_name=data.display_name,
+                password=secrets.token_urlsafe(128),
+            )
+
+            logger.info(f"User {data.display_name} - {data.email} created in FinOps for Cloud.")
+            new_user_created = True
+            user_data = response.json()
+
+    # check if the employee already belongs to the org. If not, the user will be added.
+    with wrap_http_error_in_502(
+        f"Error Adding Employee {data.display_name} "
+        f"to FinOps for Cloud Organization {db_organization.name}."
+    ):
+        response = await optscale_client.create_org_employee(
+            organization_id=db_organization.id,
+            user_id=user_data["id"],
+            name=data.display_name,
+        )
+        if response.status_code == status.HTTP_409_CONFLICT:
+            logger.info(
+                f"The User {user_data['email']} already belongs to the "
+                f"Organization {db_organization.name}."
+            )
+    with wrap_http_error_in_502(
+        f"Error Promoting User {data.display_name} "
+        f"to Admin in FinOps for Cloud {db_organization.name}."
+    ):
+        # promote the new user to admin
+        await optscale_auth_client.make_user_admin(
+            organization_id=db_organization.id,
+            user_id=user_data["id"],
+        )
+    if new_user_created:
+        # start the reset password processes for the new created user
+        with wrap_http_error_in_502(
+            "Error resetting the password for employee in FinOps for Cloud"
+        ):
+            await optscale_client.reset_password(data.email)
```

Comment:
One question. Let's assume `add_additional_admin` failed on optscale_auth_client.make_user_admin or optscale_client.reset_password. I suppose, in this case user will repeat the same request but new_user_created will be False and password won't be reset. Is it a problem?
And have you tested that calling optscale_auth_client.make_user_admin for a user that already is an admin doesn't trigger error?